ebpfcore/ebpf_drv.c

```diff
@@ -65,9 +67,126 @@ static _Function_class_(EVT_WDF_DRIVER_UNLOAD) _IRQL_requires_same_
 
     _ebpf_driver_unloading_flag = TRUE;
 
+    if (ebpf_execution_context_privileged_security_descriptor) {
+        ebpf_free(ebpf_execution_context_privileged_security_descriptor);
+        ebpf_execution_context_privileged_security_descriptor = NULL;
+    }
+
     ebpf_core_terminate();
 }
 
+static _Check_return_ NTSTATUS
+_ebpf_driver_build_privileged_security_descriptor()
+{
+    PACL dacl = NULL;
+    PSID sid = NULL;
+    NTSTATUS status = STATUS_SUCCESS;
+    SECURITY_DESCRIPTOR security_descriptor;
+    PSECURITY_DESCRIPTOR self_relative_security_descriptor = NULL;
+    const ULONG sidSubAuthorities[] = {3453964624, 2861012444, 1105579853, 3193141192, 1897355174};
+    const SID_IDENTIFIER_AUTHORITY svcAuth = {0x00, 0x00, 0x00, 0x00, 0x00, 0x50}; // S-1-5-80
+    const ULONG subauthority_count = EBPF_COUNT_OF(sidSubAuthorities);
```

Comment:
nit: assuming "subauthority" is one word, then
```suggestion
    const ULONG subauthority_count = EBPF_COUNT_OF(sidSubauthorities);
```